connectors/sources/github.py

```diff
@@ -533,22 +674,24 @@ async def get_github_item(self, resource):
         except Exception:
             raise
 
-    def get_page_info(self, response, keys):
-        """Get page information from the response.
+    def get_data_by_keys(self, response, keys, endKey):
+        """Retrieve data from a nested dictionary using a list of keys and an end key.
 
         Args:
-            response (dict): Response of GitHub
-            Keys (list): List of fields to get pageInfo
+            response (dict): The nested dictionary from which data will be extracted.
+            keys (list): A list of strings representing the keys to navigate the nested dictionary.
+            endKey (str): The final key to retrieve the desired data from the nested dictionary.
```

Comment:
why have an `endKey` and not just have that be the last key in `keys`?

Reply:
Because in `_fetch_pull_requests` and in `_fetch_issues` we are passing keys more than one time. one to get `pageInfo` and one to get `nodes`. So if we are remove the `endKey` then we need to update keys multiple times i.e. add `pageInfo` in keys then remove `pageInfo` and add `nodes` etc. So for simplicity I have added endKey argument.